lib/src/facts/bsd/networking_resolver.cc

```diff
@@ -153,6 +155,56 @@ namespace facter { namespace facts { namespace bsd {
                 return true;
             }, "^dhclient.*lease.*$");
         }
+    }
+
+    void networking_resolver::find_networkd_dhcp_servers(std::map<std::string, std::string>& servers) const
+    {
+        static const string networkd_lease_directory = "/run/systemd/netif/leases/";
```

Comment:
Files in this directory look something like this:
```
root@k1tyuo9fuykflek:~# cat /run/systemd/netif/leases/2 
# This is private data. Do not parse.
ADDRESS=10.32.119.212
NETMASK=255.255.240.0
ROUTER=10.32.112.1
SERVER_ADDRESS=10.32.22.9
NEXT_SERVER=10.28.22.13
T1=450
T2=787
LIFETIME=900
DNS=10.240.0.10 10.240.1.10
NTP=10.32.22.9 10.0.22.10 10.32.22.10 10.0.22.11
DOMAINNAME=delivery.puppetlabs.net
```
The "do not parse" message is ominous, but I wasn't sure where else to find this information. At least on Ubuntu 18.04, these files are readable by non-root users.

Reply:
That is, indeed, ominous.

I'd hope they expose this information /somewhere/ as a stable API, but who knows.

Reply:
Can you add a comment here with the info that this isn't public API, but its the best we can find